core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -24,15 +24,120 @@ coreos:
+        BindsTo=etcd-member.service
+        Before=etcd-member.service
+        Wants=cfn-etcd-environment.service
+        After=cfn-etcd-environment.service
+        Wants=network.target
+
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        RestartSec=5
+        EnvironmentFile=-/etc/etcd-environment
+        EnvironmentFile=-/var/run/coreos/etcdadm-environment
+        ExecStartPre=/usr/bin/mkdir -p /var/run/coreos/etcdadm/snapshots
+        ExecStart=/opt/bin/etcdadm reconfigure
+        TimeoutStartSec=120
+
+    - name: etcdadm-update-status.service
+      enable: true
+      content: |
+        [Unit]
+        Description=etcdadm update status
+        BindsTo=etcd-member.service
+        After=etcd-member.service
+        Wants=network.target
+
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        RestartSec=5
+        EnvironmentFile=-/etc/etcd-environment
+        EnvironmentFile=-/var/run/coreos/etcdadm-environment
+        ExecStart=/opt/bin/etcdadm member_status_set_started
+        TimeoutStartSec=120
+
+    - name: etcdadm-check.service
+      enable: true
+      content: |
+        [Unit]
+        Description=etcd health check
+        Wants=network.target
+
+        [Service]
+        Type=simple
+        EnvironmentFile=-/etc/etcd-environment
+        EnvironmentFile=-/var/run/coreos/etcdadm-environment
+        ExecStart=/opt/bin/etcdadm check
+        TimeoutStartSec=120
+
+    - name: etcdadm-check.timer
+      enable: true
+      command: start
+      content: |
+        [Unit]
+        Description=periodic etcd health check
+
+        [Timer]
+        OnBootSec=120sec
+        # Actual interval would be OnUnitInactiveSec+0~AccuracySec=10+0~5 sec
+        OnUnitInactiveSec=10sec
+        AccuracySec=5sec
+
+        [Install]
+        WantedBy=timers.target
+
+    - name: etcdadm-save.service
+      enable: true
+      content: |
+        [Unit]
+        Description=etcd snapshot
+        Wants=network.target
+
+        [Service]
+        Type=simple
+        EnvironmentFile=-/etc/etcd-environment
+        EnvironmentFile=-/var/run/coreos/etcdadm-environment
+        Restart=no
+        ExecStartPre=systemctl is-active etcd-member.service
+        ExecStart=/opt/bin/etcdadm save
+        TimeoutStartSec=300
+
+    - name: etcdadm-save.timer
+      enable: true
+      command: start
+      content: |
+        [Unit]
+        Description=periodic etcd snapshot
+
+        [Timer]
+        OnBootSec=120sec
+        # Actual interval would be 60+0~5 sec
+        OnUnitInactiveSec=60sec
+        AccuracySec=5sec
+
+        [Install]
+        WantedBy=timers.target
+
+    - name: etcd-member.service
       drop-ins:
-        - name: 20-etcd2-aws-cluster.conf
+        - name: 20-aws-cluster.conf
           content: |
             {{- if .TLSAssetsEncryptionEnabled}}
             [Unit]
             Wants=decrypt-tls-assets.service cfn-etcd-environment.service
```

Comment:
It is required to dynamically populate `/etc/etcd-environment` which is then read by etcd?